tests/test_server.py

```diff
@@ -202,8 +246,12 @@ async def test_version_removal(client, server):
         assert res.code == 200
         result = await client.get_project(id=project_id)
 
+        result = await client.get_setting(tid=env_id, id=data.AVAILABLE_VERSIONS_TO_KEEP)
+        assert result.code == 200
+        n_versions_to_keep = result.result["value"]
+
         versions = await client.list_versions(tid=env_id)
-        assert versions.result["count"] <= opt.server_version_to_keep.get() + 1
+        assert versions.result["count"] <= n_versions_to_keep + 1
```

Comment:
```suggestion
        assert versions.result["count"] == n_versions_to_keep
```
Make it so we check that we have exactly the expected number of versions